src/csharp/Microsoft.Spark/Sql/RowCollector.cs

```diff
@@ -33,7 +34,29 @@ public IEnumerable<Row> Collect(ISocketWrapper socket)
 
                 foreach (object unpickled in unpickledObjects)
                 {
-                    yield return (unpickled as RowConstructor).GetRow();
+                    // yield return (unpickled as RowConstructor).GetRow();
+                    switch (unpickled)
+                    {
+                        case RowConstructor rc:
+                            yield return rc.GetRow();
+                            break;
+
+                        case object[] objs:
+                            if ((objs.Length != 1) || !(objs[0] is Row row))
```

Comment:
Can you add comment why this is a special case? If you have two columns each with StructType, which case will handle it? (you may want to add a test for that as well).

Reply:
Do you mean the schema should look like below?
```cs
var schema = new StructType(new[]
{
    new StructField("col1", new StructType(new[]
    {
        new StructField("subCol1", new IntegerType())
    })),
    new StructField("col2", new StructType(new[]
    {
        new StructField("subCol2", new StringType())
    }))
});
```

Reply:
Nested case test has been added.

Reply:
Resolved as per discussion. Thanks!